// @uppy/components/lib/hooks/dropzone.js
const fileInputId = 'uppy-dropzone-file-input';
export function createDropzone(ctx, options = {}) {
    const handleDrop = (event) => {
        event.preventDefault();
        event.stopPropagation();
        const files = Array.from(event.dataTransfer?.files ?? []);
        if (!files.length)
            return;
        options.onDrop?.(files);
        ctx.uppy.addFiles(files.map((file) => ({
            name: file.name,
            type: file.type,
            data: file,
        })));
    };
    const handleDragOver = (event) => {
        event.preventDefault();
        event.stopPropagation();
        options.onDragOver?.(event);
    };
    const handleDragEnter = (event) => {
        event.preventDefault();
        event.stopPropagation();
        options.onDragEnter?.(event);
    };
    const handleDragLeave = (event) => {
        event.preventDefault();
        event.stopPropagation();
        options.onDragLeave?.(event);
    };
    const handleClick = () => {
        if (options.noClick)
            return;
        const input = document.getElementById(fileInputId);
        input?.click();
    };
    const handleFileInputChange = (event) => {
        const input = event.target;
        const files = Array.from(input.files ?? []);
        if (!files.length)
            return;
        options.onFileInputChange?.(files);
        ctx.uppy.addFiles(files.map((file) => ({
            source: 'drag-drop',
            name: file.name,
            type: file.type,
            data: file,
        })));
        // Reset the input value so the same file can be selected again
        input.value = '';
    };
    return {
        getRootProps: () => ({
            onDragEnter: handleDragEnter,
            onDragOver: handleDragOver,
            onDragLeave: handleDragLeave,
            onDrop: handleDrop,
            onClick: handleClick,
        }),
        getInputProps: () => ({
            id: fileInputId,
            type: 'file',
            multiple: true,
            onChange: handleFileInputChange,
        }),
    };
}
